File: BasicWebApp/src/main/java/Extention/Listener/SessionListener.java
package Extention.Listener;

import javax.servlet.ServletContext;
import javax.servlet.ServletContextEvent;
import javax.servlet.ServletContextListener;
import javax.servlet.annotation.WebListener;
import javax.servlet.http.HttpSession;
import javax.servlet.http.HttpSessionEvent;
import javax.servlet.http.HttpSessionListener;
import java.util.concurrent.atomic.AtomicInteger;

/**
 * Created by ZhangJin on 2017/7/15.
 * <p>
 * 同时继承了 HttpSessionListener 和  HttpSessionListener
 * <p>
 * HttpSessionListener 相关的方法用于注册 session 的创建及销毁方法
 */
@WebListener
public class SessionListener implements HttpSessionListener, ServletContextListener {

    public void contextInitialized(ServletContextEvent servletContextEvent) {

    }

    public void contextDestroyed(ServletContextEvent servletContextEvent) {
    }

    public void sessionCreated(HttpSessionEvent httpSessionEvent) {

    }

    public void sessionDestroyed(HttpSessionEvent httpSessionEvent) {

    }
}

File: BasicWebApp/src/main/kotlin/basicwebapp/extention/listener/SessionListener.kt
package basicwebapp.extention.listener

import javax.servlet.ServletContextEvent
import javax.servlet.ServletContextListener
import javax.servlet.annotation.WebListener
import javax.servlet.http.HttpSessionEvent
import javax.servlet.http.HttpSessionListener

/**
 * Created by ZhangJin on 2017/7/15.
 *
 *
 * 同时继承了 HttpSessionListener 和  HttpSessionListener
 *
 *
 * HttpSessionListener 相关的方法用于注册 session 的创建及销毁方法
 */
@WebListener
class SessionListener : HttpSessionListener, ServletContextListener {

    override fun contextInitialized(servletContextEvent: ServletContextEvent) {

    }

    override fun contextDestroyed(servletContextEvent: ServletContextEvent) {}

    override fun sessionCreated(httpSessionEvent: HttpSessionEvent) {

    }

    override fun sessionDestroyed(httpSessionEvent: HttpSessionEvent) {

    }
}
